pkg/login/social/generic_oauth.go

```diff
@@ -341,12 +351,27 @@ func (s *SocialGenericOAuth) extractRole(data *UserInfoJson) (string, error) {
 	}
 
 	role, err := s.searchJSONForAttr(s.roleAttributePath, data.rawJSON)
+
 	if err != nil {
 		return "", err
 	}
 	return role, nil
 }
 
+func (s *SocialGenericOAuth) extractGroups(data *UserInfoJson) ([]string, error) {
+	if s.groupsAttributePath == "" {
+		return []string{}, nil
+	}
+
+	groups, err := s.searchJSONForAttr(s.groupsAttributePath, data.rawJSON)
+
+	if err != nil && len(groups) == 0 {
+		return []string{}, err
+	}
+
+	return strings.Split(groups, ","), nil
```

Comment:
I wonder what's most common/most user friendly. Require user to create a comma-separated list or an array. My bet is that it's more common to have an array (list) of groups in JSON payloads. 

Do you have a Jmespath example of creating a comma-separated string based on an array?

WDYT?

Reply:
the reason I went for string, is that the searchJSONForAttr method was created for strings, not array. Happy to add a searchJSONForAttr for string arrays

Reply:
ah I know understand your suggestion. Yes, agreed. Let me check if JMESPATH can output comma seperated. That prevents confusion on the format.

Reply:
searchJSONForAttr for array is probably a good idea

Reply:
I've added an example for a JMESPath that returns a comma seperated string from a json array value